events/models.py

```diff
@@ -40,6 +40,15 @@ def user_can_view(self, user):
     def published(self):
         return self.filter(published_at__isnull=False)
 
+    def upcoming(self):
+        return self.filter(events__occurrences__time__gte=timezone.now()).distinct()
+
+    def upcoming_with_leeway(self):
+        return self.filter(
+            events__occurrences__time__gte=timezone.now()
+            - timedelta(minutes=settings.KUKKUU_ENROLLED_OCCURRENCE_IN_PAST_LEEWAY)
```

Comment:
What if the client could decide the amount of leeway minutes? No configuration needed then (or a default value could be added).

Reply:
Using the strict leeway setting / boolean has/had the benefit that e.g. this result will be in sync with `childNode.pastEvents`.

Reason why I used boolean filters was simply that `OccurrenceNode` already had the same kind of `upcoming` and `upcoming_with_leeway` boolean filters and I wanted to keep this similar.

Another option (or additional filter) to using a strict or UI defined leeway value could be to just return events that are in-progress and upcoming. Currently `upcoming_with_leeway` returns future events and events which have started no longer than 30 minutes ago.